integration_tests/test_stage.py

```diff
@@ -64,3 +66,26 @@ def test_staging_libc_links(self):
                 self.fail('Parts unexpectedly conflicted')
             else:
                 raise
+
+    def test_classic_confinement(self):
+        project_dir = 'classic-build'
+
+        # The first run should fail as the environment variable is not
+        # set but we can only test this on clean systems.
+        if not os.path.exists(os.path.join(
+                os.path.sep, 'snap', 'core', 'current')):
+            try:
+                self.run_snapcraft(['stage'], project_dir)
+            except subprocess.CalledProcessError as e:
+                pass
+            else:
+                self.fail(
+                    'This should fail as SNAPCRAFT_SETUP_CORE is not set')
+
+        # Now we set the required environment variable
+        self.useFixture(fixtures.EnvironmentVariable(
+                'SNAPCRAFT_SETUP_CORE', '1'))
```

Comment:
It sounds like cjwatson will be mounting the core snap into place, so perhaps we don't need to worry about setting this in LP.